added config timeout_s_read and delay_ms_baudrate_change for MeterD0

mbehr1 opened the pull request:
- added two new configurable timeouts/delays for MeterD0:
 - timeout_s_read: timeout between singles bytes read from device in seconds. Was fixed to 10s till now.
 - delay_ms_baudrate_change: delay in ms between ACKSEQ send and before baud rate change. Defaults to 0 in which case the previous (tcdrain) solution is used.
- ospeed and ispeed are now always set to same value as some adapters might have problems with different settings.

r00t- commented:
isn't "timeout_ms_baudrate_change" a delay and not a timeout?

mbehr1 commented:
you're right. I'll tonight.

Gruß
Matthias 

Sent from a mobile device. 

> Am 15.01.2015 um 00:05 schrieb r00t- <notifications@github.com>:
> 
> isn't "timeout_ms_baudrate_change" a delay and not a timeout?
> 
> —
> Reply to this email directly or view it on GitHub.
>

mbehr1 commented:
updated PR to reflect one timeout and one delay. changed parameter name and description.

nyphis commented:
I can confirm the PR works fine for me.
With 400ms delay my powermeter is sending data at 9600 baud!

2015-01-15 21:31 GMT+01:00 Matthias Behr <notifications@github.com>:

> updated PR to reflect one timeout and one delay. changed parameter name
> and description.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/volkszaehler/vzlogger/pull/111#issuecomment-70156146>.
>



-- 
Martin Heinze

ICQ: 21741119
Skype: nyphis

mbehr1 commented:
Good :-)

Can I get a dump as well to check communication?

> Am 15.01.2015 um 22:44 schrieb Martin Heinze <notifications@github.com>:
> 
> I can confirm the PR works fine for me. 
> With 400ms delay my powermeter is sending data at 9600 baud! 
> 
> 2015-01-15 21:31 GMT+01:00 Matthias Behr <notifications@github.com>: 
> 
> > updated PR to reflect one timeout and one delay. changed parameter name 
> > and description. 
> > 
> > — 
> > Reply to this email directly or view it on GitHub 
> > <https://github.com/volkszaehler/vzlogger/pull/111#issuecomment-70156146>. 
> > 
> 
> 
> 
> -- 
> Martin Heinze 
> 
> ICQ: 21741119 
> Skype: nyphis
> —
> Reply to this email directly or view it on GitHub <https://github.com/volkszaehler/vzlogger/pull/111#issuecomment-70167714>.
> 

Gruß

Matthias

mbehr1 commented:
Updated PR to add delay between baud rate change for pullseq as well.

andig commented on etc/vzlogger.conf:
```diff
@@ -62,6 +62,8 @@
             "protocol": "d0",               // see 'vzlogger -h' for list of available protocols
             "device": "/dev/ttyUSB0",
             "dump_file": "/var/log/dumpD0.txt", // optional, if set logs all received/transmitted data to this file
+//          "timeout_s_read": 10, // optional, default 10s. Timeout value in secs between single bytes received from device
```
Suggested name: read_timeout for consistency, add unit to comment

andig commented on etc/vzlogger.conf:
```diff
@@ -62,6 +62,8 @@
             "device": "/dev/ttyUSB0",
             "dump_file": "/var/log/dumpD0.txt", // optional, if set logs all received/transmitted data to this file
+//          "timeout_s_read": 10, // optional, default 10s. Timeout value in secs between single bytes received from device
+//          "delay_ms_baudrate_change": 400, // optional, default none. Delay value in ms after ACKSEQ send before baudrate change
```
Suggested name: baudrate_change_delay for consistency, add ms unit to comment

andig commented on src/protocols/MeterD0.cpp:
```diff
@@ -221,6 +223,23 @@ MeterD0::MeterD0(std::list<Option> options)
 	}
 
+    try {
+        _timeout_s_read = optlist.lookup_int(options, "timeout_s_read");
```
dito

andig commented on src/protocols/MeterD0.cpp:
```diff
@@ -221,6 +223,23 @@ MeterD0::MeterD0(std::list<Option> options)
+    }
+
+    try {
+		_delay_ms_baudrate_change = optlist.lookup_int(options, "delay_ms_baudrate_change");
```
dito

mbehr1 commented:
I changed the option names. For the variables I kept the units as this is less error prone (especially since they have different units).
I took a look at the 62056-21 spec and found one problem that could lead to timeouts. I added handling of min reaction time to keep a small delay before we send an answer to the meter.

andig commented:
Klasse Arbeit, vielen Dank!

andig commented:
Ist jetzt im Master- ganz herzlichen Dank an Matthias für die viele Arbeit!

On Sat, Jan 24, 2015 at 8:59 AM, Matthias Behr <notifications@github.com>
wrote:

> I changed the option names. For the variables I kept the units as this is
> less error prone (especially since they have different units).
> I took a look at the 62056-21 spec and found one problem that could lead
> to timeouts. I added handling of min reaction time to keep a small delay
> before we send an answer to the meter.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/volkszaehler/vzlogger/pull/111#issuecomment-71304536>.
>

mbehr1 commented:
Gern geschehen!
> Am 24.01.2015 um 19:38 schrieb andig <notifications@github.com>:
> 
> Ist jetzt im Master- ganz herzlichen Dank an Matthias für die viele Arbeit! 
> 
> On Sat, Jan 24, 2015 at 8:59 AM, Matthias Behr <notifications@github.com> 
> wrote: 
> 
> > I changed the option names. For the variables I kept the units as this is 
> > less error prone (especially since they have different units). 
> > I took a look at the 62056-21 spec and found one problem that could lead 
> > to timeouts. I added handling of min reaction time to keep a small delay 
> > before we send an answer to the meter. 
> > 
> > — 
> > Reply to this email directly or view it on GitHub 
> > <https://github.com/volkszaehler/vzlogger/pull/111#issuecomment-71304536>. 
> >
> —
> Reply to this email directly or view it on GitHub <https://github.com/volkszaehler/vzlogger/pull/111#issuecomment-71310107>.
> 

Gruß

Matthias
